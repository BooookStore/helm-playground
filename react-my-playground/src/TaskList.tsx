import { useReducer, useState } from "react";
import { initialTasks, tasksReducer } from "./tasksReducer";

export default function TaskList() {
  const [newTaskText, setNewTaskText] = useState("");
  const [tasks, dispatch] = useReducer(tasksReducer, initialTasks);

  function onAddButtonClick() {
    dispatch({ type: "ADD", text: newTaskText });
    setNewTaskText("");
  }

  function onCheckClick(id: string) {
    dispatch({ type: "DELETE", id });
  }

  return (
    <div>
      <h2>Task List</h2>
      <ul style={{ listStyleType: "none", padding: "0px" }}>
        {tasks.map((task) => (
          <li key={task.id}>
            <input
              type="checkbox"
              style={{ display: "inline" }}
              onChange={() => onCheckClick(task.id)}
            />
            <p style={{ display: "inline" }}>{task.text}</p>
          </li>
        ))}
      </ul>
      <input
        type="text"
        value={newTaskText}
        style={{ marginRight: "10px" }}
        onChange={(e) => setNewTaskText(e.target.value)}
      />
      <button onClick={onAddButtonClick}>add</button>
    </div>
  );
}
