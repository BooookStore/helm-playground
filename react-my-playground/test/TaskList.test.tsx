import React from "react";

import { describe, test, afterEach, expect } from "vitest";
import { render, cleanup } from "@testing-library/react";

import TaskList from "../src/TaskList";

describe("TaskList", () => {
  test("render", () => {
    const { container } = render(<TaskList />);
    expect(container).toMatchSnapshot();
  });

  afterEach(cleanup);
});
