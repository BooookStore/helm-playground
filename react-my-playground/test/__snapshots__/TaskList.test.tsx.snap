// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TaskList > render 1`] = `
<div>
  <div>
    <h2
      class="underline"
    >
      Task List
    </h2>
    <ul
      style="list-style-type: none; padding: 0px;"
    >
      <li>
        <input
          style="display: inline;"
          type="checkbox"
        />
        <p
          style="display: inline;"
        >
          first task
        </p>
      </li>
      <li>
        <input
          style="display: inline;"
          type="checkbox"
        />
        <p
          style="display: inline;"
        >
          second task
        </p>
      </li>
      <li>
        <input
          style="display: inline;"
          type="checkbox"
        />
        <p
          style="display: inline;"
        >
          third task
        </p>
      </li>
    </ul>
    <input
      style="margin-right: 10px;"
      type="text"
      value=""
    />
    <button>
      add
    </button>
  </div>
</div>
`;
